third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -4957,9 +4952,24 @@ void SetupRecordReplayCommands(v8::Isolate* isolate, LocalFrame* localFrame) {
   }
 }
 
-void RunInitialRecordReplayScripts(v8::Isolate* isolate) {
+void OnNewWindow(v8::Isolate* isolate, LocalFrame* localFrame) {
   v8::Local<v8::Context> context = isolate->GetCurrentContext();
 
+  // Add __RECORD_REPLAY_ANNOTATION_HOOK__ as a global.
+  SetFunctionProperty(isolate, context->Global(), AnnotationHookJSName,
+                      InvokeOnAnnotation);
+}
+
+void OnNewRootFrame(v8::Isolate* isolate, LocalFrame* localFrame) {
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  
+  // 1. Register navigation event.
+  if (localFrame->GetDocument()->Url().ProtocolIsInHTTPFamily()) {
```

Comment:
There can be navigation without creating a new frame since pages can use pushState and popState and such, so I'm not sure this covers that case.

Reply:
Hum... That's a good point. Those events will definitely not get triggered here.
I will keep this change in because its better than before because previously, it sometimes would not show any navigation events, at least we reliably get a subset of them.
However, I will need to do a follow-up to actually fix RUN-1010.